rskj-core/src/main/java/co/rsk/util/EthSwapUtil.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.util;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.core.bc.ClaimTransactionInfoHolder;
+import co.rsk.db.RepositorySnapshot;
+import org.ethereum.config.Constants;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.solidity.SolidityType;
+import org.ethereum.util.ByteUtil;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.GasCost;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+public class EthSwapUtil {
```

Comment:
Why do we call it `EthSwap`? I know that's the orig contract this is based on, but the use case is so different that this name is just confusing. Even Sergio's `ClaimTimelock` would have been better, even though that's still not descriptive enough, but at least it doesn't add confusion by referring to ETH and swap. This goes for all fields and methods, too. Unless I'm missing something?
I'd suggest something in the lines of gas deferring, or self-paying.
Perhaps:
- DefferedGasTransaction
- SelfPayingTransaction
- DefferedGasVault
- DefferedGasLocker
- GasDeferredWithdraw
- SelfPayingWithdraw

I'm sure we can think of a better name together.

Reply:
What about `ClaimTransactionValidator`? Since the class is used to validate transactions that are sent specifically to claim funds from a contract.

Reply:
addressed, changed the name to `ClaimTransactionValidator`